### PR TITLE
Feature: more function tests

### PR DESCRIPTION
**Pull request by Exanlv**
More tests woo

~~also updated cs cuz my local repo wasnt updated when I PR'd those other changes~~

### CODE REVIEW

**Review by davidcole1340 — DISMISSED** [review]
thanks!

**Review by key2peace — APPROVED** [review]

**Comment by SQKo** [comment]
These tests should be in separated workflow to reduce github actions time, since the result is not based the Discord API

File: tests/FunctionsTest.php [hunk 1]
```diff
@@ -63,10 +81,147 @@ public function strlenProvider()
...
     /**
-     * @dataProvider strlenProvider
+     * @test
+     * @dataProvider mentionedProvider
      */
-    public function testPolyStrlen($expected, $string)
+    public function testMentioned($part, $attributes, $outcome): void
     {
-        $this->assertEquals($expected, poly_strlen($string));
+        $mockDiscord = getMockDiscord();
```
Comment (SQKo) [inline tests/FunctionsTest.php:89]: I'm not sure why the same code is not used elsewhere

Comment (Exanlv) [reply tests/FunctionsTest.php:89]: Not sure what you mean by this

Comment (SQKo) [reply tests/FunctionsTest.php:89]: such as
```php
return [
	[new User(getMockDiscord(), ['id' => '12345']), '12345'],
	[new Channel(getMockDiscord(), ['id' => '12345']), '12345'],
	[new Role(getMockDiscord(), ['id' => '12345']), '12345'],
	['12345', '12345'],
	[null, null],
];
```

```
$mockDiscord = getMockDiscord();

return [
	[new User($mockDiscord, ['id' => '12345']), '12345'],
	[new Channel($mockDiscord, ['id' => '12345']), '12345'],
	[new Role($mockDiscord, ['id' => '12345']), '12345'],
	['12345', '12345'],
	[null, null],
];
```

Comment (Exanlv) [reply tests/FunctionsTest.php:89]: Tbh I never thought about adding more logic then just a single return in a `dataProvider`

**Review by SQKo — APPROVED** [review]
Will think about the separate workflow later

**Pull request merged**